_episodes/01-introduction.md

```diff
@@ -66,31 +66,72 @@ formalism.
   in terms of tasks or workflows. This is `dask.delayed`.
 
 # What is parallel computing?
-Suppose we have a computation, where each step **depends** on a previous one:
+
+## Dependency diagrams
+
+Suppose we have a computation, where each step **depends** on a previous one. We can represent this situation like in the figure below, known as a dependency diagram:
+
+![Serial computation](../fig/serial.png)
+
+In these diagrams the inputs and outputs of each function are specified as inward and outward arrows. Note that the output of one function can become the input of another one. The diagram above is the typical diagram of a **serial computation**. If you ever used a loop to update a value, you used serial computation.
+
+If our computation involves **independent work** (that is, the results of the application of each function are independent of the results of the application of the rest), we can structure our computation like this:
+
+![Parallel computation](../fig/parallel.png)
+
+This scheme corresponds to a **parallel computation**.
+
+### How can parallel computing improve our code execution speed?
+
+Nowadays, most personal computers have 4 or 8 processors (also known as cores). In the diagram above, we can assign each of the three functions to one core, so they can be performed simultaneously.
+
+It may be tempting to think that using three cores instead of one would multiply the execution speed by three. For now, it's ok to use this a first approximation to reality. Later in the course we'll see that things are actually more complicated than that.
+
+## Parallelizable and non-parallelizable tasks
+
+It is important to know that some tasks are fundamentally non-parallelizable, also known as **inherently serial**. An example could be the brute-force computation of the factorial of an integer (example: the factorial of 4, usually written as $4!$ is obtained by multiplying all the integers smaller or equal to $6$, that is: $4! = 4 \cdot 3 \cdot 2 \cdot 1 = 24$ ).
```

Comment:
The formulas here do not render properly. (I think the notation with $ only works in R-markdown).

Reply:
I've committed a fix (but I am no expert in this, I just followed https://stackoverflow.com/a/47798853/4711118

Reply:
If I remember properly, we had a similar issue before. We fixed it by adding [this snippet](https://github.com/escience-academy/lesson-template/commit/fcb625142bdbf1b4c746d99640de21a6e07e8a5c).

Reply:
Applied the snippet in commit 7c94d59608b1a15c3af4ad4c016c56cbf7d27ce3.